pkg/transport/filters/access_type_filter.go

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package filters
+
+import (
+	"fmt"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+)
+
+type AccessTypeFilterSpec struct {
+	IncludeAccessTypes []string `json:"includeAccessTypes"`
+}
+
+type accessTypeFilter struct {
+	includeAccessTypes []string
+}
+
+func (f accessTypeFilter) Matches(cd cdv2.ComponentDescriptor, r cdv2.Resource) bool {
+	for _, accessType := range f.includeAccessTypes {
+		if r.Access.Type == accessType {
+			return true
+		}
+	}
```

Comment:
wouldnt a map/set make more sense if you want to know if a string is part of the set?

Reply:
obviously yes :) I've updated the code